apps/explorer/src/pages/transaction-result/SendReceiveView.tsx

```diff
@@ -1,21 +1,115 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 import cl from 'clsx';
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useContext } from 'react';
 
 import { ReactComponent as DoneIcon } from '../../assets/SVGIcons/16px/CheckFill.svg';
 import { ReactComponent as StartIcon } from '../../assets/SVGIcons/Start.svg';
 import Longtext from '../../components/longtext/Longtext';
+import { NetworkContext } from '../../context';
+import { DefaultRpcClient as rpc } from '../../utils/api/DefaultRpcClient';
+import { parseObjectType } from '../../utils/objectUtils';
 
 import styles from './SendReceiveView.module.css';
 
 type TxAddress = {
     sender: string;
     recipient?: string[];
     amount?: bigint[];
+    objects?: string[];
 };
+
+function MultipleRecipients({ sender, recipient, amount, objects }: TxAddress) {
+    const [network] = useContext(NetworkContext);
+
+    const [coinList, setCoinList] = useState({
+        loadState: 'pending',
+        data: [],
+    });
+
+    useEffect(() => {
+        if (objects) {
+            Promise.all(
```

Comment:
Using `useQuery` you should be able to simplify this down to:

```ts
const networkObjects = await Promise.all(
  objects.map((objId) => rpc(network).getObject(objId))
);

return networkObjects.map((obj) => parseObjectType(obj));
```

You can also just use the `enabled` flag to make it conditional on `objects` existing.